tests/unit/test_git_url_watcher.py

```diff
@@ -88,6 +90,110 @@ def git_config(branch_name: str, git_repository: Callable[[], str]) -> dict[str,
     return cfg
 
 
+@pytest.fixture()
+def git_config_two_repos_synced_same_tag_regex(
+    branch_name: str, git_repository: Callable[[], str]
+) -> dict[str, Any]:
+    cfg = {
+        "main": {
+            "synced_via_tags": True,
+            "watched_git_repositories": [
+                {
+                    "id": "test-repo-" + str(i),
+                    "url": str(git_repository()),
+                    "branch": branch_name,
+                    "tags": "^staging_.+",
+                    "paths": [],
+                    "username": "",
+                    "password": "",
+                }
+                for i in range(2)
+            ],
+        }
+    }
+    return cfg
+
+
+async def test_git_url_watcher_tag_sync(
+    event_loop, git_config_two_repos_synced_same_tag_regex: dict[str, Any]
+):
+    repo_id_var: str = git_config_two_repos_synced_same_tag_regex["main"][
+        "watched_git_repositories"
+    ][0]["id"]
+    branch_var: str = git_config_two_repos_synced_same_tag_regex["main"][
+        "watched_git_repositories"
+    ][0]["branch"]
+    local_path_var: str = git_config_two_repos_synced_same_tag_regex["main"][
+        "watched_git_repositories"
+    ][0]["url"].replace("file://localhost", "")
+
+    assert git_config_two_repos_synced_same_tag_regex["main"]["synced_via_tags"]
+    git_watcher = git_url_watcher.GitUrlWatcher(
+        git_config_two_repos_synced_same_tag_regex
+    )
+    with pytest.raises(ConfigurationError):
+        init_result = await git_watcher.init()
+
+    # add a file, commit, and tag
+    VALID_TAG: Literal["staging_z1stvalid"] = "staging_z1stvalid"
+    TESTFILE_NAME: Literal["testfile.csv"] = "testfile.csv"
+    for repo in [
+        git_config_two_repos_synced_same_tag_regex["main"]["watched_git_repositories"][
+            i
+        ]
+        for i in range(
+            len(
+                git_config_two_repos_synced_same_tag_regex["main"][
+                    "watched_git_repositories"
+                ]
+            )
+        )
+    ]:
+        _run_cmd(
+            f"touch {TESTFILE_NAME}; git add .; git commit -m 'pytest: I added {TESTFILE_NAME}'; git tag {VALID_TAG};",
+            cwd=repo["url"].replace("file://localhost", ""),
+        )
+        assert await git_url_watcher._check_if_tag_on_branch(
+            repo["url"].replace("file://localhost", ""), branch_var, VALID_TAG
+        )
+    init_result = await git_watcher.init()
+    assert not await git_watcher.check_for_changes()
+    time.sleep(1.1)
```

Comment:
TIP: also some tenacity retry pattern in this test would be nice as well

Reply:
thanks, will add this possibly in a later refactoring :)